### ISSUE TITLE
Install library with namespace

### ISSUE THREAD

**Issue by MichaelUnknown** [TheLartians/PackageProject.cmake#9]
Hi,
I think an option to export a namespace to my library would be nice (or I totally missed how to do it).
So with my current cmake file
```
add_library(mkpoker INTERFACE)
add_library(mkpoker::mkpoker ALIAS mkpoker)
```
I would like to do this
```
set(MY_LIBRARY_NAMESPACE "mkpoker")
add_library(${MY_LIBRARY_NAMESPACE}::mkpoker ALIAS mkpoker)

packageProject(
    NAME ${PROJECT_NAME}
    VERSION ${PROJECT_VERSION}
    LIBRARY_NAMESPACE ${MY_LIBRARY_NAMESPACE}
    BINARY_DIR ${PROJECT_BINARY_DIR}
    INCLUDE_DIR ${PROJECT_SOURCE_DIR}/include
    INCLUDE_DESTINATION include/${PROJECT_NAME}-${PROJECT_VERSION}
    VERSION_HEADER "${PROJECT_NAME}/version.h"
)
```
so that the cmake install adds the lib with the correct namespace, resulting in something like
```
add_library(mkpoker::mkpoker INTERFACE IMPORTED)
```

Best wishes,
mike

PS: I am confused, how `DEPENDENCIES` is supposed to work... if I add my librarys dependencies with CPM.cmake, this will install all of them regardless, i.e., even if no deps are specified in the `DEPENDENCIES`  variable

**Comment by TheLartians** [comment]
Hey, thanks for the suggestion, that's a great idea! I'll check it out as soon as I have some time, but I'm also happy to accept a PR. 😉 

> PS: I am confused, how DEPENDENCIES is supposed to work... if I add my librarys dependencies with CPM.cmake, this will install all of them regardless, i.e., even if no deps are specified in the DEPENDENCIES variable

The difference is that `find_pacakge(mpoker)` will also find any specified dependencies. Otherwise the library user would need to manually search for all dependencies before adding the library. See [Config.cmake.in](https://github.com/TheLartians/PackageProject.cmake/blob/master/Config.cmake.in) and the definition of the [FindDependency macro](https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html) for details.

**Comment by MichaelUnknown** [comment]
> 
> 
> Hey, thanks for the suggestion, that's a great idea! I'll check it out as soon as I have some time, but I'm also happy to accept a PR. 😉

I think I can reverse engineer what is needed by looking at other project's install cmake files. I will maybe give it a try this evening.

**Comment by MichaelUnknown** [comment]
resolved with #10
